### PR TITLE
Do not accept all paths not starting with 'login'

### PR DESCRIPTION
**Pull request by millerdev**
I feel dirty for not writing tests for this...

@czue cc @proteusvacuum

### CODE REVIEW

File: corehq/apps/hqwebapp/middleware.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def _reject(self, request, reason):
         _assert = soft_assert('{}@{}'.format('sreddy+logs', 'dimagi.com'), exponential_backoff=False)
         _assert(False, warning)
 
-        if request_url.find('login'):
```
Comment (millerdev) [inline corehq/apps/hqwebapp/middleware.py]: `str.find(sub)` returns `-1` if `sub` is not found and `bool(-1) == True`

**Comment by czue** [comment]
ouch

File: corehq/apps/hqwebapp/middleware.py [hunk 2]
```diff
@@ -27,7 +27,7 @@ def _reject(self, request, reason):
...
+        if '/login/' in request_url:
```
Comment (millerdev) [inline corehq/apps/hqwebapp/middleware.py]: Don't like this check. It's possibly too permissive. The previous check (if it had been phrased more correctly as something like `'login' in request_url`) is trivially exploited by creating a domain with the letters 'login' embedded somewhere in the name. I think that's no longer possible, but I'm not confident that I can imagine all similar attack vectors to know that the new check is completely safe either. Would be better to do a positive match on known login URL patterns, but I'm not sure how to do that easily.

Why is it important that we disable CSRF protection on login URLs?

**Comment by czue** [comment]
> Why is it important that we disable CSRF protection on login URLs?

I was wondering this too. @sravfeyn ?

**Comment by millerdev** [comment]
>> Why is it important that we disable CSRF protection on login URLs?
>
> I was wondering this too. @sravfeyn ?

@czue can we merge this anyway? In any case this is more correct than what we had before.

**Comment by czue** [comment]
yes, good call. @sravfeyn when you're back can you let us know the rationale behind this?

**Comment by sravfeyn** [comment]
We are not supposed to disable csrf on login pages. Some of the login requests (that are correct) were failing on CSRF. I couldn't figure out why they were failing, so I exempted them _temporarily_ (so that we can do ICDS audit) till I can look at that issue again. I have this on my backlog.

Change from this PR looks good and is better than old. Thanks Daniel!

**Comment by czue** [comment]
@sravfeyn are you sure they were actually failing from valid login attempts? could have been random botspam type of stuff since those pages are public.

**Comment by sravfeyn** [comment]
They seem to be valid requests that are supposed to succeed. I will create an FB ticket and ping you with the log for FYI

**Pull request merged**